crates/publish/src/expander.rs

```diff
@@ -370,13 +370,26 @@ fn bindle_id(
     app_info: &local_schema::RawAppInformation,
     buildinfo: Option<BuildMetadata>,
 ) -> PublishResult<bindle::Id> {
+    let safe_name = safe_bindle_name(&app_info.name);
     let text = match buildinfo {
-        None => format!("{}/{}", app_info.name, app_info.version),
-        Some(buildinfo) => format!("{}/{}+{}", app_info.name, app_info.version, buildinfo),
+        None => format!("{}/{}", safe_name, app_info.version),
+        Some(buildinfo) => format!("{}/{}+{}", safe_name, app_info.version, buildinfo),
     };
     bindle::Id::try_from(&text).map_err(|_| PublishError::BindleId(text))
 }
 
+// This is both slightly conservative and slightly loose. According to the spec,
+// the / character should also be allowed, but currently that is not supported
+// by Hippo (Spin issue 504). And the - character should not be allowed, but lots of
+// our tests use it...!
+lazy_static::lazy_static! {
+    static ref UNSAFE_CHARACTERS: regex::Regex = regex::Regex::new("[^-_\\p{L}\\p{N}]").expect("Invalid name regex");
+}
+
+fn safe_bindle_name(name: &str) -> String {
```

Comment:
I think it might be easier both from a codebase upkeep perspective and to understand for the end user if we rejected unsafe names rather than transforming them into safe names. What do you think @itowlson ? Are there other reasons for transforming the name on behalf of the user?

Reply:
Sure!  I'll make that change - thanks.